askama_parser/src/lib.rs

```diff
@@ -487,3 +511,77 @@ fn filter(i: &str, level: Level) -> ParseResult<'_, (&str, Option<Vec<Expr<'_>>>
     ))(i)?;
     Ok((i, (fname, args)))
 }
+
+/// Returns the common parts of two paths.
+///
+/// The goal of this function is to reduce the path length based on the `base` argument
+/// (generally the path where the program is running into). For example:
+///
+/// ```text
+/// current dir: /a/b/c
+/// path:        /a/b/c/d/e.txt
+/// ```
+///
+/// `strip_common` will return `d/e.txt`.
+fn strip_common(base: &Path, path: &Path) -> String {
+    let path = match path.canonicalize() {
+        Ok(path) => path,
+        Err(_) => return path.display().to_string(),
+    };
+    let mut components_iter = path.components().peekable();
+
+    for current_path_component in base.components() {
+        let Some(path_component) = components_iter.peek() else {
+            return path.display().to_string();
+        };
+        if current_path_component != *path_component {
+            break;
+        }
+        components_iter.next();
+    }
+    let path_parts = components_iter
+        .map(|c| c.as_os_str().to_string_lossy())
+        .collect::<Vec<_>>();
+    if path_parts.is_empty() {
+        path.display().to_string()
+    } else {
+        path_parts.join("/")
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::strip_common;
+    use std::path::Path;
+
+    #[test]
+    #[cfg(not(windows))]
```

Comment:
Why can't we run this test on Windows? Feels like it might actually be important that we test this bit of platform-sensitive logic on Windows, too?

Reply:
Because the top component of a path on Windows seems to be multiple potential things, even if it's all the same (take a look at [Component::Prefix](https://doc.rust-lang.org/nightly/std/path/enum.Component.html#variant.Prefix) documentation for some extra info.

But in short, when generating the canonicalized path, it generated `\\C:\\thing` whereas normally you would expect `C:\\thing`. Based on that, I simply decided to postpone it for a later time to someone who actually cares about Windows. ^^'

Reply:
So what is the behavior going to be on Windows? Maybe adding these error locations is actually a good reason that we should have UI tests enabled on Windows.

Reply:
> So what is the behavior going to be on Windows? 

Keeping the full path for most cases (which isn't the best but still better than no path).

> Maybe adding these error locations is actually a good reason that we should have UI tests enabled on Windows.

That wouldn't work because of paths `/` vs `\\`. Or we would need to have two sets of ui tests.